### PR TITLE
Issue #537: Generate .vscode recommandations file

### LINKED ISSUE #537: Generate .vscode recommandations file
### Description
As the adapters group is progressing, it should be good that the zip returned would contain .vscode recommandations files so that user will be suggested to install VSCode extensions relevant to the project (ex Java LSP, OpenShift tooling,...).
@mohitsuman would be able to provide the content of those files. Depending on the target date, the vscode fabric8 launcher extension could generate those files if the folder is not present in the zip (see fabric8-launcher/launcher-vscode-extension#10
...

### What should happen?
.vscode folder added to the zip with the relevant file(s)

...

### What happens instead?
No .vscode folder yet
...

### Do you have any idea why?

...

### Error log

```
If you have an error log, please paste it here.
```

### PR DESCRIPTION
**Pull request by jeffmaury**
Signed-off-by: Jeff MAURY <jmaury@redhat.com>

### Why

To provide VSCode recommandations files

### Description

Will add VSCode recommandation file to the generated project in order to let user know which VSCode extensions are good for him

Closes #537.

### Checklist

- [x] I followed the contribution [guidelines](https://raw.githubusercontent.com/fabric8-launcher/launcher-backend/master/CONTRIBUTING.md).
- [x] I created an [issue](https://github.com/fabric8-launcher/launcher-backend/issues/new) and used it in the commit message.
- [ ] I have built the project locally prior to submission with `mvn clean install`.
- [x] I kept a clean commit log (no unnecessary commits, no merge commits).
- [x] I am happy with my contribution.

### CODE REVIEW

File: core/core-impl/src/main/java/io/fabric8/launcher/core/impl/preparers/VSCodePreparer.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    public void prepare(Path projectPath, RhoarBooster booster, ProjectileContext context) {
+        if (!(context instanceof BoosterCapable)) {
+            return;
+        }
+        BoosterCapable createProjectileContext = (BoosterCapable) context;
+        generateVSCodeRecommandations(projectPath, createProjectileContext.getRuntime().getId());
+    }
+
+    void generateVSCodeRecommandations(Path projectPath, String runtimeId) {
```
Comment (gastaldi) [inline core/core-impl/src/main/java/io/fabric8/launcher/core/impl/preparers/VSCodePreparer.java]: s/recommandations/recommendations

**Comment by jeffmaury** [comment]
Yes but Che will soon be VSCode compatible :)

**Comment by ALRubinger** [comment]
@jeffmaury Cool stuff.  But you see my point, yes?  This is a great value-add and I'd like to be sure we do it in a way that's consistent with the IDEs we offer as part of the Red Hat Development Experience.

**Comment by gorkem** [comment]
@jeffmaury che will not be compatible with `.vscode` folder. It is plugin API level compatibility only.

File: core/core-impl/src/main/java/io/fabric8/launcher/core/impl/preparers/VSCodePreparer.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                    Files.createDirectories(projectPath.resolve(VSCODE_FOLDER));
+                    Files.copy(stack, projectPath.resolve(VSCODE_FOLDER).resolve(EXTENSIONS_JSON_FILE));
+                }
+            } catch (IOException e) {
+                throw new UncheckedIOException("Error while traversing project " + projectPath, e);
+            }
+        }
+    }
+
+    private boolean needsVSCodeRecommandations(Path projectPath) {
```
Comment (gastaldi) [inline core/core-impl/src/main/java/io/fabric8/launcher/core/impl/preparers/VSCodePreparer.java]:
```suggestion
    private boolean needsVSCodeRecommendations(Path projectPath) {
```

File: core/core-impl/src/main/java/io/fabric8/launcher/core/impl/preparers/VSCodePreparer.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    public void prepare(Path projectPath, RhoarBooster booster, ProjectileContext context) {
+        if (!(context instanceof BoosterCapable)) {
+            return;
+        }
+        BoosterCapable createProjectileContext = (BoosterCapable) context;
+        generateVSCodeRecommandations(projectPath, createProjectileContext.getRuntime().getId());
+    }
+
+    void generateVSCodeRecommandations(Path projectPath, String runtimeId) {
+        if (needsVSCodeRecommandations(projectPath)) {
```
Comment (gastaldi) [inline core/core-impl/src/main/java/io/fabric8/launcher/core/impl/preparers/VSCodePreparer.java]:
```suggestion
        if (needsVSCodeRecommendations(projectPath)) {
```

File: core/core-impl/src/main/java/io/fabric8/launcher/core/impl/preparers/VSCodePreparer.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    public static final String EXTENSIONS_JSON_FILE = "extensions.json";
+    public static final String VSCODE_FOLDER = ".vscode";
+
+    @Override
+    public void prepare(Path projectPath, RhoarBooster booster, ProjectileContext context) {
+        if (!(context instanceof BoosterCapable)) {
+            return;
+        }
+        BoosterCapable createProjectileContext = (BoosterCapable) context;
+        generateVSCodeRecommandations(projectPath, createProjectileContext.getRuntime().getId());
```
Comment (gastaldi) [inline core/core-impl/src/main/java/io/fabric8/launcher/core/impl/preparers/VSCodePreparer.java]:
```suggestion
        generateVSCodeRecommendations(projectPath, createProjectileContext.getRuntime().getId());
```

**Pull request merged**